torchgeo/datasets/geo.py

```diff
@@ -416,7 +416,7 @@ def __init__(
         self,
         paths: Path | Iterable[Path] = 'data',
         crs: CRS | None = None,
-        res: float | None = None,
+        res: tuple[float, float] | None = None,
```

Comment:
If possible, I would like all of these datasets to support both `float` and `tuple[float, float]` similar to how our Samplers work. We can use the same helper function to convert the former to the latter. This is also nice for backwards compatibility (which we have to start caring about soon).

Reply:
Also need to update docstrings to document the order of `xres` and `yres`.

Reply:
Which docstrings would you like this in?

Reply:
Every single docstring for every single GeoDataset with a `res` parameter.